providers/file/file.go

```diff
@@ -120,9 +121,18 @@ func (f *File) Watch(cb func(event interface{}, err error)) error {
 			}
 		}
 
-		w.Close()
+		f.w.Close()
 	}()
 
 	// Watch the directory for changes.
-	return w.Add(fDir)
+	return f.w.Add(fDir)
+}
+
+func (f *File) UnWatch() error {
+	fDir, _ := filepath.Split(f.path)
+	err := f.w.Remove(fDir)
```

Comment:
Maybe we should instead use `f.w.Close()`? This should close the channel and stop the Watch goroutine as well?

https://pkg.go.dev/github.com/fsnotify/fsnotify#Watcher.Close

Reply:
Using Close function